cli/client.go

```diff
@@ -1197,3 +1209,45 @@ func (c *viamClient) startRobotPartShell(
 	outputLoop()
 	return nil
 }
+
+func parseLogEntryFields(fieldsRaw []*structpb.Struct) ([]zap.Field, error) {
+	parsedFields := make([]zap.Field, 0, len(fieldsRaw))
+	bytes, err := json.Marshal(fieldsRaw)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(bytes, &parsedFields); err != nil {
+		return nil, err
+	}
+	return parsedFields, nil
+}
+
+func logEntryFieldsToString(fields []zap.Field) string {
+	fieldsString := ""
+	for _, field := range fields {
+		fieldValue := ""
+		// TODO use zap encoder to encode
+		// it is a bit hard to do that though because sending the data over the wire
+		// loses type info which causes the zap encoders to panic.
+		// This code is modeled after zapcore.Field.AddTo
+		//nolint:exhaustive
+		switch field.Type {
+		case zapcore.StringType, zapcore.ErrorType:
+			fieldValue = field.String
+		case zapcore.BoolType:
+			fieldValue = fmt.Sprint(field.Integer == 1)
+		case zapcore.DurationType:
+			fieldValue = time.Duration(field.Integer).String()
+		case zapcore.Int64Type, zapcore.Int32Type, zapcore.Int16Type, zapcore.Int8Type:
+			fieldValue = fmt.Sprint(field.Integer)
+		case zapcore.Float64Type:
+			fieldValue = fmt.Sprint(math.Float64frombits(uint64(field.Integer)))
+		case zapcore.Float32Type:
+			fieldValue = fmt.Sprint(math.Float32frombits(uint32(field.Integer)))
+		default:
+			fieldValue = fmt.Sprint(field.Interface)
+		}
+		fieldsString += fmt.Sprintf("\t%s %s", field.Key, fieldValue)
```

Comment:
@dgottlieb do we prefer tab-separated key-value pairs like this? Or, something like `{"key": "value", "key2": 1}` instead? I'm fine with either, but want to align how we print fields for the CLI, modules and app.

Reply:
We should go with `{"key": "value", "key2": 1}`.

Reply:
^ agree with that

Reply:
Fixed in [77a0c1f](https://github.com/viamrobotics/rdk/pull/3613/commits/77a0c1f4daf40e9f7c4300e0799a21e4d4e01909)